HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -458,15 +468,17 @@ public async Task<IActionResult> Eligible_Post(EligibleViewModel viewModel)
     }
     
     [HttpGet("confirmation")]
-    public IActionResult Confirmation_Get()
+    public IActionResult Confirmation_Get(bool emailPreferenceSubmitted)

```

Comment:
I did this as a query param rather than a property on the questionnaire so that if the user goes back and forward they still see both versions of the page.